src/Streamly/Internal/Data/Fold.hs

```diff
@@ -416,6 +417,46 @@ scan (Fold stepL initialL extractL) (Fold stepR initialR extractR) =
 
     extract = extractR . snd
 
+-- | Postscan the input of a 'Fold' to change it in a stateful manner using
+-- another 'Fold'.
+-- /Pre-release/
+{-# INLINE postscan #-}
+postscan :: Monad m => Fold m a b -> Fold m b c -> Fold m a c
+postscan (Fold stepL initialL extractL) (Fold stepR initialR extractR) =
+    Fold step initial extract
+
+    where
+
+    runStep actionL sR = do
```

Comment:
Since this is common in `scan` and `postscan`, can we take this out and reuse in both?

Reply:
It is a good idea to add an INLINE annotation on this function.

Reply:
> Since this is common in `scan` and `postscan`, can we take this out and reuse in both?

I'm facing some difficulties in making the `runStep` function (taken outside) to typecheck.
```
 Couldn't match expected type ‘s3’ with actual type ‘s1’
      ‘s3’ is a rigid type variable bound by
        a pattern with constructor:
          Fold :: forall (m :: * -> *) a b s.
                  (s -> a -> m (Step s b))
                  -> m (Step s b) -> (s -> m b) -> Fold m a b,
        in an equation for ‘runStep’
        at src/Streamly/Internal/Data/Fold.hs:388:41-68
      ‘s1’ is a rigid type variable bound by
        the type signature for:
          runStep :: forall (m :: * -> *) a b c s s1.
                     Monad m =>
                     Fold m a b
                     -> Fold m b c -> m (Step s b) -> s1 -> m (Step (s, s1) c)
        at src/Streamly/Internal/Data/Fold.hs:387:1-111
        ```

Reply:
You can leave it like that, just add a to-do comment saying that these two are the same can use a common definition.